packages/graph/hash_graph/lib/graph/src/ontology/data_type.rs

```diff
@@ -151,13 +148,20 @@ pub struct DataTypeQueryPathVisitor {
     position: usize,
 }
 
+impl DataTypeQueryPathVisitor {
+    #[must_use]
+    pub const fn new(position: usize) -> Self {
+        Self { position }
+    }
+}
+
 impl<'de> Visitor<'de> for DataTypeQueryPathVisitor {
     type Value = DataTypeQueryPath<'de>;
 
     fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
         formatter.write_str(
-            "a sequence containing one element of `ownedById`, `baseUri`, `versionedUri`, \
-             `version`, `title, `description`, `type`, or a custom identifier",
+            "one of `ownedById`, `baseUri`, `versionedUri`, `version`, `title, `description`, \
```

Comment:
This seems fairly easy to fall out of sync. I don't really have a suggestion but is there any way for us to make this more automatic, lintable, or type safe?

Reply:
We could use a crate like [`strum`](https://crates.io/crates/strum) and iterate over the variants. I don't know if it's worth it, though.

Reply:
_Probably_ not worth it. Let's keep it in mind if we do find ourselves making mistakes with this.